### PR TITLE
support categorical_column_identity and embedding column

### PR DESCRIPTION
**Pull request by uuleon**
1. support categorical_column_identity and embedding column in `codegen_alps`
2. refactoring the unit test of `codegen_alps`
3. bugfix

### CODE REVIEW

File: sql/codegen_alps.go [hunk 1]
```diff
@@ -27,6 +27,8 @@ const (
 	sparse    = "SPARSE"
 	numeric   = "NUMERIC"
 	cross     = "CROSS"
+	catID     = "CAT_ID"
```
Comment (wangkuiyi) [inline sql/codegen_alps.go:30]: catID => categoryId
CAT_ID => CATEGORY_ID

File: sql/codegen_alps.go [hunk 2]
```diff
@@ -90,6 +92,17 @@ type crossColumn struct {
 	HashBucketSize int
 }
 
+type catIDColumn struct {
```
Comment (wangkuiyi) [inline sql/codegen_alps.go:95]: catIDColumn => categoricalIdColumn

**Comment by wangkuiyi** [comment]
Looks generally good to me. Could @tonyyang-svail please have a look?

**Review by tonyyang-svail — APPROVED** [review]
LGTM.

**Comment by tonyyang-svail** [comment]
I have a couple of design questions regarding the following code generated by `TestTrainALPSFiller.wndStatement`.

```python
class SQLFlowEstimatorBuilder(EstimatorBuilder):
  def _build(self, experiment, run_config):
    return tf.estimator.DNNLinearCombinedClassifier(
      dnn_hidden_units=[10,20],
      config=run_config,dnn_feature_columns=[
        tf.feature_column.numeric_column("dense", shape=(5,)),
        tf.feature_column.embedding_column(
          tf.feature_column.categorical_column_with_identity(key="deep", num_buckets=2000),
          dimension=8, combiner="mean")
        ],
        linear_feature_columns=[
          tf.feature_column.embedding_column(
            tf.feature_column.categorical_column_with_identity(key="wide", num_buckets=1000),
            dimension=16, combiner="mean")
        ]
      )

if __name__ == "__main__":
  trainDs = DatasetX(
    reader=OdpsReader(
      odps=OdpsConf(
        accessid=None,
        accesskey=None,
        endpoint=None
      ),
      project=None,
      table="kaggle_credit_fraud_training_data",
      field_names=["dense","deep","wide"],
      features=[DenseColumn(name="dense", shape=[5], dtype="float", separator=","),
      SparseColumn(name="deep", shape=[2000], dtype="float", separator=","),
      SparseColumn(name="wide", shape=[1000], dtype="float", separator=",")],
      labels=DenseColumn(name="c3", shape=[1], dtype="int", separator=",")
    )
  )

  evalDs = DatasetX(
    num_epochs=1,
    batch_size=64,
    reader=OdpsReader(
      odps=OdpsConf(
        accessid=None,
        accesskey=None,
        endpoint=None
      ),
      project=None,
      table="kaggle_credit_fraud_training_data",
      field_names=["dense","deep","wide"],
      features=[DenseColumn(name="dense", shape=[5], dtype="float", separator=","),
      SparseColumn(name="deep", shape=[2000], dtype="float", separator=","),
      SparseColumn(name="wide", shape=[1000], dtype="float", separator=",")],
      labels=DenseColumn(name="c3", shape=[1], dtype="int", separator=",")
    )
  )

  export_path = "/tmp/alps_scratch_dir_780774028/model/"

  experiment = Experiment(
    user="sqlflow",
    engine=LocalEngine(),
    train=TrainConf(input=trainDs,max_steps=1000,),
    eval=EvalConf(input=evalDs,),
    exporter=ArksExporter(deploy_path=export_path,
      strategy=ExportStrategy.BEST,
      compare_fn=Closure(best_auc_fn)),
    model_dir="/tmp/alps_scratch_dir_780774028",
    model_builder=SQLFlowEstimatorBuilder())

  run_experiment(experiment)
```

1. Is it possible to make `DenseColumn` and `SparseColumn` part of the feature column's API? I can think of two benefits:
    1. Since feature column is part of the model definition, different data readers can share these preprocessing columns.
    1. We can avoid repetitions on field names and shapes such as 
        1. ` tf.feature_column.numeric_column("dense", shape=(5,))` at model definition and
        1. `DenseColumn(name="dense", shape=[5], dtype="float", separator=",")` at `datasetX` definition
1. Is it possible to make `datasetX` support reading data from a query? In the near future, SQLFlow will support arbitary select statement prior to the train statement. It is likely the select statement involves more than one table.

**Pull request merged**